<robot name="OmuniRobot" xmlns:xacro="http://ros.org/wiki/xacro">
  <xacro:macro name="imu_macro" params="parent prefix xyz">
    <joint name="imu_joint" type="fixed">
      <parent link="${parent}"/>
      <child  link="${prefix}_link"/>
      <origin xyz="${xyz}" rpy="0 0 0"/>
    </joint>
    <link name="${prefix}_link">
      <visual>
        <origin xyz="0 0 0" rpy="0 0 0" />
        <geometry>
          <box size="0.015 0.025 0.005" />
        </geometry>
        <material name="blue" />
      </visual>
    </link>
    <gazebo reference="${prefix}_link">
      <gravity>true</gravity>
      <sensor name="imu_sensor" type="imu">
        <always_on>true</always_on>
        <update_rate>100</update_rate>
        <visualize>true</visualize>
        <topic>data</topic>
        <plugin filename="libgazebo_ros_imu_sensor.so" name="imu_plugin">
          <topicName>${prefix}/data</topicName>
          <bodyName>${prefix}_link</bodyName>
          <updateRateHZ>10.0</updateRateHZ>
          <gaussianNoise>0.0</gaussianNoise>
          <xyzOffset>0 0 0</xyzOffset>
          <rpyOffset>0 0 0</rpyOffset>
          <frameName>${prefix}_link</frameName>
        </plugin>
        <pose>0 0 0 0 0 0</pose>
      </sensor>
    </gazebo>
  </xacro:macro> 
</robot>
